staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/schemas.go

```diff
@@ -0,0 +1,418 @@
+			"intVal": {Name: "intVal", Type: IntType},
+		})
+	}
+	if s.XEmbeddedResource {
+		return NewObjectType("embedded", map[string]*DeclField{
+			"kind":       {Name: "kind", Type: StringType},
+			"apiVersion": {Name: "apiVersion", Type: StringType},
+			"metadata": {Name: "metadata", Type: NewObjectType("metadata", map[string]*DeclField{
+				"name": {Name: "name", Type: StringType},
+			})},
+			"spec":   {Name: "spec", Type: DynType},
+			"status": {Name: "status", Type: DynType, Required: false},
+		})
+	}
+	if s.XPreserveUnknownFields {
+		return DynType
+	}
+	switch declType.TypeName() {
+	case ListType.TypeName():
+		return NewListType(SchemaDeclType(s.Items))
+	case MapType.TypeName():
+		if s.AdditionalProperties != nil && s.AdditionalProperties.Structural != nil {
+			return NewMapType(StringType, SchemaDeclType(s.AdditionalProperties.Structural))
+		}
+		fields := make(map[string]*DeclField, len(s.Properties))
+
+		required := map[string]bool{}
+		if s.ValueValidation != nil {
+			for _, f := range s.ValueValidation.Required {
+				required[f] = true
+			}
+		}
+		for name, prop := range s.Properties {
+			var enumValues []interface{}
+			if prop.ValueValidation != nil {
+				for _, e := range prop.ValueValidation.Enum {
+					enumValues = append(enumValues, e.Object)
+				}
+			}
+			fields[Escape(name)] = &DeclField{
+				Name:         Escape(name),
+				Required:     required[name],
+				Type:         SchemaDeclType(&prop),
+				defaultValue: prop.Default.Object,
+				enumValues:   enumValues, // TODO: escape?
+			}
+		}
+		return NewObjectType("object", fields)
+	case StringType.TypeName():
+		if s.ValueValidation != nil {
+			switch s.ValueValidation.Format {
+			case "byte":
+				return StringType // OpenAPIv3 byte format represents base64 encoded string
+			case "binary":
+				return BytesType
+			case "google-duration":
+				return DurationType
+			case "date", "date-time", "google-datetime":
+				return TimestampType
+			case "int64":
+				return IntType
+			case "uint64":
+				return UintType
+			}
+		}
+	}
+	return declType
+}
+
+var (
+	// SchemaDef defines an Open API Schema definition in terms of an Open API Schema.
+	schemaDef *schema.Structural
+
+	// AnySchema indicates that the value may be of any type.
+	AnySchema *schema.Structural
+
+	// EnvSchema defines the schema for CEL environments referenced within Policy Templates.
+	envSchema *schema.Structural
+
+	// InstanceSchema defines a basic schema for defining Policy Instances where the instance rule
+	// references a TemplateSchema derived from the Instance's template kind.
+	instanceSchema *schema.Structural
+
+	// TemplateSchema defines a schema for defining Policy Templates.
+	templateSchema *schema.Structural
+
+	openAPISchemaTypes = map[string]*DeclType{
+		"boolean":         BoolType,
+		"number":          DoubleType,
+		"integer":         IntType,
+		"null":            NullType,
+		"string":          StringType,
+		"google-duration": DurationType,
```

Comment:
Were you planning on adding support for `int-or-string` at this level, or in some other file? I think I might have lost the context where the type support is being added among the file set.

Reply:
Yes, the support has been added to schemas.go `SchemaDeclType` func so:

```
declType, found := openAPISchemaTypes[s.Type]
	if !found {
		return nil
	}
	if s.XIntOrString {
		return NewObjectType("intOrString", map[string]*DeclField{
			"strVal": {Name: "strVal", Type: StringType},
			"intVal": {Name: "intVal", Type: IntType},
		})
	}
```

I'm also dealing with embedded resources and "preserve unknown fields" there.